soroban-env-host/src/host/conversion.rs

```diff
@@ -485,7 +485,27 @@ impl Host {
                     let v = self.to_host_val(&pair.val)?;
                     mm.push((k, v))
                 }
-                Ok(self.add_host_object(HostMap::from_map(mm, self)?)?.into())
+                Ok(self
+                    .add_host_object(HostMap::from_map(mm, self).map_err(|err| {
+                        let mut err = err;
+                        self.with_debug_mode(|| {
+                            if !err.error.is_type(ScErrorType::Object)
+                                || !err.error.is_code(ScErrorCode::InvalidInput)
```

Comment:
I'm not sure I understand why these these conditions are necessary or sufficient to know "it's unsorted"

Reply:
That's how from_map is implemented. I realize that's a bit brittle, so I moved the error decoration code to metered_map.rs, so that at least it lives near the code that produces the error. I've also noticed that we create maps in two flows actually, so the helper function is useful at least for deduplication.